File: server/Models/ApiResponse.cs
namespace Api.Models
{
    public class ApiResponse<T>
    {
        public bool Success { get; set; } = true;
        public T? Data { get; set; }
        public string? ErrorMessage { get; set; }

        public static ApiResponse<T> Ok(T data) => new() { Success = true, Data = data };
        public static ApiResponse<T> Fail(string message) => new() { Success = false, ErrorMessage = message };
    }

}

File: client/types/api-response.ts
// 假設你在這個檔案裡定義，並且有從別處 import DeepSeekResponse 型別
import type { DeepSeekResponse } from "./DeepSeekRespones"; // <-- 這邊改成你實際路徑

export type ApiResponse<T> = {
  success: boolean;
  data: T | null;
  errorMessage?: string;
  errorCode?: number;
  errorDetail?: string;
};

export interface AskResultData {
  aiResponse: DeepSeekResponse;
  userMessageId: number;
  aiMessageId: number;
}
